### ISSUE TITLE
gpus_ReturnNotPermittedKillClient error occuring in iOS background-aware applications 

### ISSUE THREAD

**Issue by msmollin** [tangrams/tangram-es#1708]
We received some feedback in the Mapzen iOS SDK causing crashes for applications going into the background. This seems to stem from GLKitViewController not pausing its rendering (like I thought it did) when going into and out of background. On reflection this does make some sense as it would be silly for Apple to assume it knows the state of the render pipeline.

To resolve this, we should follow the guidelines outlined in https://developer.apple.com/library/content/documentation/3DDrawing/Conceptual/OpenGLES_ProgrammingGuide/ImplementingaMultitasking-awareOpenGLESApplication/ImplementingaMultitasking-awareOpenGLESApplication.html#//apple_ref/doc/uid/TP40008793-CH5-SW1

The tl;dr; of that document is we should subscribe to the notifications coming out of UIApplication (https://developer.apple.com/documentation/uikit/uiapplication) in TGMapViewController. That way, when the application is changing state into the background or coming into the foreground, we can  pause/resume the rendering thread appropriately. We should also dispose of any resources we don't need on certain events.

**Comment by msmollin** [comment]
I've reproduced this error inside the Tangram-es sample app for iOS. I'll be opening a PR later today or tomorrow with cleaned up code to help test this and subsequent fixes. The basic problem is the Mapzen iOS SDK is receiving location updates while in the background, and thus issues marker position animation requests, which Tangram-es is attempting to render. This makes iOS rather annoyed and causes it to emit the mentioned error (and subsequently crash).

So to reproduce it in the sample app you must:

- Add a few keys to the samples info plist:
-- Background mode for location updates
-- Privacy keys for both Always and When in use, as well as just for When in Use (unsure why we need both here but iOS insists upon it)
- Link against CoreLocation
- Import CoreLocation in the sample app
- Add a `CLLocationManager` and set its delegate to the demo view controller, and setup authorization states for a "Always in Use" authorization
- Add a new Marker that has a class property so we can update it in separate function scopes
- Configure the CLLocationManager to issue location updates when in the background using high accuracy so the system attempts to give us location updates as often as possible
- Optionally add observers to know when the app is going into and coming out of background states.

The fix is to have a way to exit early (maybe a boolean check in platform.cpp) from the render loop when the app is in the background?

**Comment by matteblair** [comment]
Resolved by #1716
